### PR TITLE
Log daemons' failure reasons to supervisord log files

### LINKED ISSUE #21433: Log daemons' failure reasons to supervisord log files
Forked from: https://github.com/kubernetes/kubernetes/issues/21261#issuecomment-185445866

In scripts managed by supervisord, we always the output, for example, in kubelet-checker.sh:

```
  if ! curl --insecure -m ${max_seconds} -f -s https://127.0.0.1:{{kubelet_port}}/healthz > /dev/null; then
    echo "kubelet failed!"
    exit 2
```

We never know why the kubelet failed the healthcheck here.  I will provide a fix shortly.

### PR DESCRIPTION
**Pull request by dchen1107**
Fixed #21433 

Also collect etcd log for diagnose ##21206, and supervisord/kubelet\* logs for diagnose #21261

cc/ @yujuhong

### CODE REVIEW

File: test/e2e/core.go [hunk 1]
```diff
@@ -62,6 +63,8 @@ func CoreDump(dir string) {
 		cmds = append(cmds, []command{
 			{"cat /var/log/kubelet.log", "kubelet"},
 			{"cat /var/log/supervisor/supervisord.log", "supervisord"},
+			{"cat /var/log/supervisor/kubelet-stdout.log", "supervisord-kubelet-stdout"},
+			{"cat /var/log/supervisor/kubelet-stderr.log", "supervisord-kubelet-stderr"},
```
Comment (yujuhong) [inline test/e2e/core.go]: Do we also want to gather `/var/log/docker.log` and `docker-stdout/stderr.log`? I think `/var/log/docker.log` will be helpful.

Comment (dchen1107) [inline test/e2e/core.go]: ah, forgot to pick up docker.log. Will add them. I couldn't make docker-stdout/stderr.log have more information. All I can get is 

```
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
```

**Comment by dchen1107** [comment]
Also collected docker.log on all nodes now.

**Comment by dchen1107** [comment]
@k8s-bot test this please github issue: #IGNORE (something happened with jenkins)

**Comment by dchen1107** [comment]
@k8s-bot test this issue: #19732

**Comment by dchen1107** [comment]
Last failure is also related to #19732

**Comment by brendandburns** [comment]
@k8s-bot test this please.  issue: #19732

**Comment by yujuhong** [comment]
@k8s-bot test this. issue: #21467

File: test/e2e/core.go [hunk 2]
```diff
@@ -62,7 +63,10 @@ func CoreDump(dir string) {
...
 			{"cat /var/log/kern.log", "kern.log"},
+			{"cat /var/log/docker.log", "docker.log"},
```
Comment (ixdy) [inline test/e2e/core.go]: don't need the `.log` in the second string - it gets added automatically. i.e. this should be
```go
{"cat /var/log/docker.log", "docker"},
```

(similar for other uses. `kern.log` is likewise incorrect, though I missed that earlier)

**Comment by brendandburns** [comment]
I'm going to just merge this...

**Pull request merged**